src/index.html

```diff
@@ -14,6 +14,68 @@
   </head>
   <body>
     <h1 class="logo">MA</h1>
-    <script src="scripts/main.js"></script>
+    <div id="notifications"></div>
+    <script>
+      function createNotification(message, type) {
+        const notification = document.createElement('div');
+        notification.className = type;
+        notification.dataset.qa = 'notification';
+        notification.textContent = message;
+        document.getElementById('notifications').appendChild(notification);
+      }
+
+      const firstPromise = new Promise((resolve, reject) => {
+        let clicked = false;
+        document.addEventListener('click', (event) => {
+          if (event.button === 0) {
+            clicked = true;
+            resolve('First promise was resolved');
+          }
+        });
+
+        setTimeout(() => {
+          if (!clicked) {
+            reject('First promise was rejected');
+          }
+        }, 3000);
```

Comment:
The firstPromise might not work as expected if a click event is not triggered within 3 seconds. This is because the promise will be rejected, but the event listener is not removed, potentially leading to memory leaks or unexpected behavior if the click event is triggered later.